website_chd_product_configurator/controllers/controllers.py

```diff
@@ -0,0 +1,237 @@
+                # extrapolate the id encoded the name
+                accessory_id = int(key.split('_')[1])
+                # get the associated value by choosing the field with the
+                # right name accessoryid_{id}=on/off and the associated
+                # quantity would be qtyaccessoryid_{id}=9898
+                accessory_qty = form_data['qty' + key]
+                new_accessory = request.env['chd.accessoire_line'].create({
+                    'product_id': accessory_id,
+                    'configurator_id': new_chd.id,
+                    'quantity': accessory_qty,
+                    })
+                all_accessories.append(new_accessory)
+        # our product configurator is ready, we can now calculate options
+        # _model refers to old API model, self.pool
+        # is not available in controller context (praise the lord for Holger!)
+        try:
+            res = new_chd._model.calculate_price(
+                request.cr, request.uid, [new_chd.id], context=request.context)
+        except orm.except_orm:
+            errormsg = "No result found for the values that you entered. \
+            We would be happy to give you a custom quote. \
+            Please call 010-7856766"
+        results = chd_results.search([('configurator_id', '=', new_chd.id)])
+        if errormsg != "" and len(results.ids) == 0:
+            chd_types = request.env['product.type']
+            Conf_products = request.env['product.template']
+            accessories = request.env['product.product']
+            curr_types = chd_types.search(
+                [('product_option_ids', 'in', [int(form_data['product_id'])])])
+            avail_accessories = accessories.search(
+                [('id', 'in', curr_product_id.chd_accessoire_ids.ids)])
+            curr_chd_price_component_ats = chd_price_components_at.search(
+                [
+                    ('id', 'in', curr_product_id.attribute_template_ids.ids),
+                    ('active', '=', True)
+                    ])
+            return request.render(
+                'website_chd_product_configurator.configurator', {
+                    'curr_product_id': curr_product_id,
+                    'curr_types': curr_types,
+                    'avail_accessories': avail_accessories,
+                    'curr_chd_price_component_ats':
+                    curr_chd_price_component_ats,
+                    'errormsg': errormsg,
+                    })
+
+        return request.render(
+            'website_chd_product_configurator.sale_options',
+            {
+                'curr_product_id': curr_product_id,
+                'curr_chd': new_chd,
+                'all_accessories': all_accessories,
+                'results': results,
+                'configuration_form': str(form_data),
+                'message': message,
+                'summary': form_data['summary'],
+                })
+
+    def get_current_partner(self):
```

Comment:
How is this different from `request.env.user.partner_id`?

If it's different, you should depend on whatever module `user_account_id` comes from